File: Problems/001-two-sum/two_sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        hash_map = {}

        for ind, val in nums:
            hash_map[val] = ind

        for ind1, val in nums:
            if target-val in hash_map:
                ind2 = hash_map[target-val]
                if ind2 != ind1:
                    return [ind1, ind2]

File: Problems/001-two-sum/two_sum.go
package problem001

func twoSum(nums []int, target int) []int {
	index := make(map[int]int, len(nums))

	for ind, val := range nums {
		index[val] = ind
	}

	for ind1, val := range nums {
		if ind2, ok := index[target-val]; ok {
			if ind2 != ind1 {
				return []int{ind1, ind2}
			}
		}
	}

	return nil
}
